src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -62,12 +64,24 @@ class ClusterObserver : IDisposable
             _cts = CancellationTokenSource.CreateLinkedTokenSource(token);
 
             var config = server.Configuration.Cluster;
-            SupervisorSamplePeriod = config.SupervisorSamplePeriod.AsTimeSpan;
-            _stabilizationTime = (long)config.StabilizationTime.AsTimeSpan.TotalMilliseconds;
+            _supervisorSamplePeriod = config.SupervisorSamplePeriod.AsTimeSpan;
+            _stabilizationTime = config.StabilizationTime.AsTimeSpan;
+            _stabilizationTimeMs = (long)config.StabilizationTime.AsTimeSpan.TotalMilliseconds;
             _moveToRehabTime = (long)config.MoveToRehabGraceTime.AsTimeSpan.TotalMilliseconds;
             _breakdownTimeout = config.AddReplicaTimeout.AsTimeSpan;
             _hardDeleteOnReplacement = config.HardDeleteOnReplacement;
-            _observe = Run(_cts.Token);
+
+            _observe = PoolOfThreads.GlobalRavenThreadPool.LongRunning(_ =>
+            {
+                try
+                {
+                    Run(_cts.Token);
+                }
+                catch
+                {
+                   // nothing we can do here
```

Comment:
You have to at least log this.

Reply:
Any exception within the observer loop will be caught and logged inside the loop itself.
If we catch anything here, it probably means logging error or some race on the cancellation token.

Reply:
This is the case right now, but it might not be there in the future. We don't try to protect ourselves from logging errors, note.